pkg/infra/baseagent.go

```diff
@@ -85,8 +85,7 @@ func (a *Base) ID() string {
 }
 
 func (a *Base) IsAlive() bool {
-	_, found := a.tower.agents[a.id]
-	return found
+	return a.hp > 0
```

Comment:
Don't think this will work since a.hp is never set to 0 I think

Reply:
V true hahaa